flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -588,23 +590,74 @@ void markFinished() {
 
 			if (current == RUNNING || current == DEPLOYING) {
 
+				ScheduleMode scheduleMode = getVertex().getExecutionGraph().getScheduleMode();
+
+				// Get the successors
+				final List<JobVertexID> successors = getVertex().getJobVertex()
+						.getJobVertex().getBatchSuccessors();
```

Comment:
Maybe these functions names should then be renamed or the list of successors should be retrieved in two steps where one sees the different types.